### DISCUSSION TITLE
谈谈 MVX 中的 Controller · /mvx-controller

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#256] [Announcements]
https://draveness.me/mvx-controller

**Comment by jackdebugging** [comment]
如果将控制器自身持有的View改成我们自己新建的新的视图对象（otherView）时，在.h文件中暴露出otherView中的一些UIButton或者其他控件，在控制器的Block中使用RAC的@weakify(self)以及@strongify(self)会造成循环引用。

````
viewController.h

@weakify(self);
self.block = ^(){
    @strongify(self);
    self.view.button = doSomething;
}

````

想问下有没有什么好的解决办法吗?

**Comment by yangjunpeng1993** [comment]
> 视图和模型之间解耦是通过控制器来处理的，控制器获取模型对象并取出其中的属性一一装填到视图中，也就是将 -setupWithModel: 方法中的代码从视图层移到控制器层中，并在视图类中暴露合适的接口。

如果这样做，填充的这部分代码就不能复用了呀？

**Comment by dimingchan** [comment]
厉害
